Allow designated coordinators to view a table of approved/sent users for their partner (T170795)

Samwalton9 opened the pull request:
I know we'd talked about using a reporting tool, but not sure how far that progressed, so put something simple together!

jsnshrmn commented on TWLight/resources/templates/resources/partner_users.html:
```diff
@@ -0,0 +1,86 @@
+  {% if approved_applications|length > 0 %}
+    {% comment %} Translators: On the page listing approved users for a partner, this is the title of the section listing applications with the 'approved' status {% endcomment %}
+    <h2>{% trans "Approved applications" %}</h2>
+    <table class="table table-sm">
```
the text in this table should be marked for translation. Have you looked at this on a phone? Make use of a responsive grid instead of a table wherever possible.

jsnshrmn commented on TWLight/resources/templates/resources/partner_users.html:
```diff
@@ -0,0 +1,86 @@
+  {% if sent_applications|length > 0 %}
+    {% comment %} Translators: On the page listing approved users for a partner, this is the title of the section listing applications with the 'sent' status {% endcomment %}
+    <h2>{% trans "Sent applications" %}</h2>
+    <table class="table table-sm">
```
the text in this table should be marked for translation. Have you looked at this on a phone? Make use of a responsive grid instead of a table wherever possible.

jsnshrmn commented on TWLight/view_mixins.py:
```diff
@@ -118,6 +119,46 @@ def dispatch(self, request, *args, **kwargs):
 
 
 
+class PartnerCoordinatorOnly(object):
```
Seems like we could have this and the coordinator or self class share more code. Perhaps a shared function for partner coordinators or maybe a class with a few subclasses?

Samwalton9 commented:
Thanks!

- Made those strings translatable (can't believe I keep forgetting to do that)
- Added a `table-responsive` div to the table, per https://getbootstrap.com/docs/4.0/content/tables/#responsive-tables. Tested with Chrome's mobile view and it seems to be working much better, table is specifically horizontally scrollable, and the header doesn't get messed up at small widths anymore.
- Your point about consolidating the mixins made me realise the partner coordinator check could just slip right in to the existing `CoordinatorOrSelf`. Tested and seems to be working fine.

jsnshrmn commented:
So does that mean that editors can now download their own data on partner pages? That seems like it wouldn't be a problem, though it might be a bit of clutter for them. Also is partner_coordinator_only in twlight_perms.py being used anywhere anymore?

Samwalton9 commented:
> So does that mean that editors can now download their own data on partner pages?

No - CoordinatorOrSelf is a bit misleading there. It checks if the view's get_object() output is an Editor, Application, or (now) Partner and corresponds to the user accordingly. Since the new view returns a Partner, the editor self check doesn't run. I've added this to the test though.

And nope, removed that function.

jsnshrmn commented:
Looks great!
